packages/zoe/src/objArrayConversion.js

```diff
@@ -14,8 +22,9 @@ export const arrayToObj = (array, keys) => {
     array.length === keys.length,
     details`array and keys must be of equal length`,
   );
-  /** @type {{[Keyword: U]: T}} */
-  const obj = {};
+  const obj =
+    /** @type {Record<U, T>} */
+    ({});
   keys.forEach((key, i) => (obj[key] = array[i]));
   return obj;
```

Comment:
I'd really like to keep property assignment out of Jessie (with the possible exception of numerically indexed arrays, which is not relevant here). I think in all cases in this file, avoiding property assignment improves the code anyway.

```suggestion
  return Object.fromEntries(keys.map((key, i) => [key, array[i]]));
```